docs/decisions/0007-release-process.md

```diff
@@ -0,0 +1,334 @@
+   git push upstream s3gw-v0.11
+   ```
+
+10. With all sub-project repositories ready to be released, it's time to prepare
+    the `s3gw` repository for a release. Much like what we did for the
+    `s3gw-charts` repository, we will have to apply a patch on the `main` branch
+    first, and then backport it to the release branch: this time to keep track
+    of the `CHANGELOG`.
+
+    First step, we need to go to the [Current CHANGELOG][8] page on the
+    repository's Wiki, and copy the contents for the release version being
+    handled to a file at `docs/release-notes/s3gw-vX.Y.Z.md`. Keep in mind the
+    release notes should be easily consumable by a human. Feel free to take
+    inspiration on previous release notes, and maintain consistency with them.
+
+    Creating a Pull Request against `s3gw`'s `main` branch is the next step.
+    Once that has been merged, note down the new commit's `SHA`, go back to the
+    `s3gw-vX.Y` branch, and cherry-pick the commit, tagging the branch with for
+    our specific release.
+
+    ```shell
+    git cherry-pick -x -s -S <SHA>
+    git tag --annotate --sign -m "Release v0.11.0" s3gw-v0.11.0
+    git push upstream s3gw-v0.11
+    ```
+
+    By pushing the branch with the release tag, we will trigger the release
+    workflow that will build the various release artifacts and publish the
+    containers on Quay.
+
+11. The last step is to officially release the new version. During the release
+    workflow, a [release draft][9] will be created. Once the release artifacts
+    have finished building, and have been published on Quay, we can then copy
+    the contents of the release notes file we created in step `10.`, and make
+    the release draft public.
+
+    It is advised that before making the release draft public, the list in
+    [Sanity Checks](#sanity-checks) be ensured to hold true.
+
+    With the release now complete, it is time to shout about it from the
+    rooftops.
+
+[^1]:
+    For example, for a `v0.11.0` release, for the `s3gw`
+    repository, go to the [Branches Page][6] and click the `New branch` button.
+
+[^2]: For example, for a `v0.11.0` release, `git branch --copy main s3gw-v0.11`
+[^3]: Please refer to [Git's Documentation][7] for more information on Tagging.
+
+[6]: https://github.com/aquarist-labs/s3gw/branches
+[7]: https://git-scm.com/book/en/v2/Git-Basics-Tagging
+[8]: https://github.com/aquarist-labs/s3gw/wiki/Current-CHANGELOG
+[9]: https://github.com/aquarist-labs/s3gw/releases
+
+### [Sanity Checks](sanity-checks)
+
+- [ ] `s3gw` container has been published on Quay for `vX.Y.Z`.
+- [ ] `s3gw-ui` container has been published on Quay for `vX.Y.Z`.
+- [ ] both containers are appropriately tagged with `vX.Y.Z` on Quay.
+- [ ] both containers are tagged with `latest` on Quay.
+- [ ] `latest` version containers are the same as the `vX.Y.Z` containers on
+      Quay.
+- [ ] Helm Chart has been properly updated for `vX.Y.Z`.
```

Comment:
And the helm chart for version `vX.Y.Z` is visible on artifacthub.io. It takes about 20 min for a correctly released chart to be picked up by artifacthub.io